src/components/badge/badge.test.tsx

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import Badge from "./badge.component";
+
+const renderComponent = (props = {}) =>
+  render(<Badge {...props}>Button</Badge>);
```

Comment:
```suggestion
  render(<Badge {...props}>Foo</Badge>);
```

non-blocking just looks a bit weird having that as the child text imo